src/Symfony/Component/Form/Extension/Core/Type/TimezoneType.php

```diff
@@ -72,7 +116,7 @@ public function loadChoiceList($value = null)
             return $this->choiceList;
         }
 
-        return $this->choiceList = new ArrayChoiceList($this->getTimezones(), $value);
+        return $this->choiceList = new ArrayChoiceList($this->getTimezones($this->regions), $value);
```

Comment:
@yceruto is right, we now need a dynamic `TimezonesChoiceLoader` implementation taking the option in its constructor.